toolchest_client/tools/tool.py

```diff
@@ -114,13 +114,16 @@ def _prepare_inputs(self):
         if self.is_database_update:
             if self.database_primary_name:
                 self.database_primary_name = os.path.basename(self.database_primary_name)
-                # If only one input file exists, use that file explicitly as the DB
-                if self.num_input_files == 1 and not self.compress_inputs:
-                    self.database_primary_name = os.path.basename(self.input_files[0])
-                # If DB name isn't a prefix for an input file, (implicitly) use directory of inputs instead as DB
-                input_basenames = [os.path.basename(input_path) for input_path in self.input_files]
-                if all([self.database_primary_name not in basename for basename in input_basenames]):
-                    self.database_primary_name = None
+                # Sanitize database_primary_name if all files are local
+                # (If there is an S3 URI, it could be a prefix, so we skip overwriting database_primary_name)
```

Comment:
Couldn't you have a prefix for a db name with local files?

Reply:
Local prefix support hasn't been implemented yet. If we pass in a local directory, it will be parsed into its individual files found along the search path (before reaching this point, IIRC).

Not sure if we want to support true "prefixes" locally in this PR or at all, which is a discussion worth having.

Reply:
I think if the use passes a prefix as a primary name we need to support it and not override it.

Reply:
Which could be true for the single input case as well.